File: python/506.py
class Solution:
    def findRelativeRanks(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """
        new_lst = sorted(nums)
        l = len(nums)
        d = {}
        ans = [""] * l
        for i in range(l):
            d[new_lst[i]] = l - i

        for j in range(l):
            if d[nums[j]] == 1:
                ans[j] = "Gold Medal"
            elif d[nums[j]] == 2:
                ans[j] = "Silver Medal"
            elif d[nums[j]] == 3:
                ans[j] = "Bronze Medal"
            else:
                ans[j] = str(d[nums[j]])
        return ans

File: java/506.java
class Solution {
    public String[] findRelativeRanks(int[] nums) {
        int[] newNum = nums.clone();
        int l = nums.length;
        Arrays.sort(newNum);
        Map<Integer,Integer>map = new HashMap();
        for(int i = 0;i < nums.length;i++){
            map.put(newNum[i],l-i-1);
        }
        String[] ans = new String[nums.length];
        for(int i = 0;i < nums.length;i++){
            if(map.get(nums[i]) == 0){
                ans[i] = "Gold Medal";
            }else if(map.get(nums[i]) == 1){
                ans[i] = "Silver Medal";
            }else if(map.get(nums[i]) == 2){
                ans[i] = "Bronze Medal";
            }else{
                ans[i] = (map.get(nums[i])+1) + "";
            }
        }
        return ans;
    }
}